In [5]:
import pandas as pd

In [7]:
import matplotlib.pyplot as plt

In [9]:
games=pd.read_csv('/Users/sharoonkk/Desktop/DA/Matplotib/Online_game.csv')

In [11]:
games

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

### Description

* This dataset appears to represent video game sales data, containing information about 16,598 video games. 
* Each row corresponds to a specific video game, and the dataset includes the following columns
   * Rank: A unique integer identifier that ranks the video games based on a specific criterion (likely sales    * Name: The name of the video ga
    * Platform: The gaming platform on which the video game was released (e.g., PS4, Xbox One, 
    * Year: The ye game was released. This column has some missing values, with 16,327 non-null entries out of 16,598.
    * Genre: The genre of the video game (e.g., Action, Sports, RPG).
    * Publisher: The company that published the video game. There are a few missing values, with 16,540 non-null entries.
    * NA_Sales: Sales in North America, measured in millions of units.
    * EU_Sales: Sales in Europe, measured in millions of units.
    * JP_Sales: Sales in Japan, measured in millions of units.
    *
Other_Sales: Sales in regions other than North America, Europe, and Japan, measured in millions of units.
    * Global_Sales: The total global sales of the video game, measured in millions of units.ar the

In [20]:
games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


In [22]:
games.isnull().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

In [102]:
games.fillna(0,inplace=True)

In [26]:
games.isnull().sum()

Rank            0
Name            0
Platform        0
Year            0
Genre           0
Publisher       0
NA_Sales        0
EU_Sales        0
JP_Sales        0
Other_Sales     0
Global_Sales    0
dtype: int64

In [13]:
games.duplicated().sum()

0

In [30]:
games.describe()

Rank          Year      NA_Sales      EU_Sales      JP_Sales  \
count  16598.000000  16598.000000  16598.000000  16598.000000  16598.000000   
mean    8300.605254   1973.647307      0.264667      0.146652      0.077782   
std     4791.853933    254.346809      0.816683      0.505351      0.309291   
min        1.000000      0.000000      0.000000      0.000000      0.000000   
25%     4151.250000   2003.000000      0.000000      0.000000      0.000000   
50%     8300.500000   2007.000000      0.080000      0.020000      0.000000   
75%    12449.750000   2010.000000      0.240000      0.110000      0.040000   
max    16600.000000   2020.000000     41.490000     29.020000     10.220000   

        Other_Sales  Global_Sales  
count  16598.000000  16598.000000  
mean       0.048063      0.537441  
std        0.188588      1.555028  
min        0.000000      0.010000  
25%        0.000000      0.060000  
50%        0.010000      0.170000  
75%        0.040000      0.470000  
max       10.570000     82.740000

In [15]:
games.describe().T

count         mean          std      min      25%      50%  \
Rank          16598.0  8300.605254  4791.853933     1.00  4151.25  8300.50   
Year          16327.0  2006.406443     5.828981  1980.00  2003.00  2007.00   
NA_Sales      16598.0     0.264667     0.816683     0.00     0.00     0.08   
EU_Sales      16598.0     0.146652     0.505351     0.00     0.00     0.02   
JP_Sales      16598.0     0.077782     0.309291     0.00     0.00     0.00   
Other_Sales   16598.0     0.048063     0.188588     0.00     0.00     0.01   
Global_Sales  16598.0     0.537441     1.555028     0.01     0.06     0.17   

                   75%       max  
Rank          12449.75  16600.00  
Year           2010.00   2020.00  
NA_Sales          0.24     41.49  
EU_Sales          0.11     29.02  
JP_Sales          0.04     10.22  
Other_Sales       0.04     10.57  
Global_Sales      0.47     82.74

In [17]:
games

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

In [19]:
games['Publisher'].unique()

array(['Nintendo', 'Microsoft Game Studios', 'Take-Two Interactive',
       'Sony Computer Entertainment', 'Activision', 'Ubisoft',
       'Bethesda Softworks', 'Electronic Arts', 'Sega', 'SquareSoft',
       'Atari', '505 Games', 'Capcom', 'GT Interactive',
       'Konami Digital Entertainment',
       'Sony Computer Entertainment Europe', 'Square Enix', 'LucasArts',
       'Virgin Interactive', 'Warner Bros. Interactive Entertainment',
       'Universal Interactive', 'Eidos Interactive', 'RedOctane',
       'Vivendi Games', 'Enix Corporation', 'Namco Bandai Games',
       'Palcom', 'Hasbro Interactive', 'THQ', 'Fox Interactive',
       'Acclaim Entertainment', 'MTV Games', 'Disney Interactive Studios',
       nan, 'Majesco Entertainment', 'Codemasters', 'Red Orb', 'Level 5',
       'Arena Entertainment', 'Midway Games', 'JVC', 'Deep Silver',
       '989 Studios', 'NCSoft', 'UEP Systems', 'Parker Bros.', 'Maxis',
       'Imagic', 'Tecmo Koei', 'Valve Software', 'ASCII Entertainment',


## 1 Retrieve the geners by Activision

In [438]:
act=games[games['Publisher']=='Activision'][['Genre','Publisher']]
act

Genre   Publisher
29     Shooter  Activision
31     Shooter  Activision
33     Shooter  Activision
34     Shooter  Activision
35     Shooter  Activision
...        ...         ...
16411   Action  Activision
16444   Action  Activision
16501  Shooter  Activision
16537   Action  Activision
16595   Racing  Activision

[975 rows x 2 columns]

In [440]:
act.nunique()

Genre        12
Publisher     1
dtype: int64

## 2 retrieve the total no of sales by Nintendo in Europe

In [122]:
g=games['EU_Sales'][games['Publisher']=='Nintendo'].max()

In [124]:
g

29.02

## 3 Retrieve the game that has highest sale in 2008

In [134]:
max=games.groupby(['Year'])

In [138]:
max=ma.get_group(2010)
max

/var/folders/n8/js2blnd547x1vm4qzc05tq5c0000gn/T/ipykernel_2295/3818359229.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  max=max.get_group(2010)


Rank                                               Name Platform  \
15        16                                 Kinect Adventures!     X360   
26        27                        Pokemon Black/Pokemon White       DS   
31        32                            Call of Duty: Black Ops     X360   
40        41                            Call of Duty: Black Ops      PS3   
54        55                                     Gran Turismo 5      PS3   
...      ...                                                ...      ...   
16437  16440              The Cages: Pro Style Batting Practice      Wii   
16478  16481                 Ishin no Arashi: Shippuu Ryuumeden       DS   
16487  16490          Slotter Mania P: Tetsuya Shinjuku vs Ueno      PSP   
16501  16504  Transformers: War for Cybertron (XBox 360, PS3...       PC   
16596  16599                                         Know How 2       DS   

         Year         Genre                     Publisher  NA_Sales  EU_Sales  \
15     2010.0          Misc        Microsoft Game Studios     14.97      4.94   
26     2010.0  Role-Playing                      Nintendo      5.57      3.28   
31     2010.0       Shooter                    Activision      9.67      3.73   
40     2010.0       Shooter                    Activision      5.98      4.44   
54     2010.0        Racing   Sony Computer Entertainment      2.96      4.88   
...       ...           ...                           ...       ...       ...   
16437  2010.0        Sports  Konami Digital Entertainment      0.01      0.00   
16478  2010.0      Strategy                    Tecmo Koei      0.00      0.00   
16487  2010.0          Misc                        Dorart      0.00      0.00   
16501  2010.0       Shooter                    Activision      0.01      0.00   
16596  2010.0        Puzzle                      7G//AMES      0.00      0.01   

       JP_Sales  Other_Sales  Global_Sales  
15         0.24         1.67         21.82  
26         5.65         0.82         15.32  
31         0.11         1.13         14.64  
40         0.48         1.83         12.73  
54         0.81         2.12         10.77  
...         ...          ...           ...  
16437      0.00         0.00          0.01  
16478      0.01         0.00          0.01  
16487      0.01         0.00          0.01  
16501      0.00         0.00          0.01  
16596      0.00         0.00          0.01  

[1259 rows x 11 columns]

## 4 Publisher with highest sale in EUROPE

In [161]:
games.groupby(['Publisher'])['EU_Sales'].max().idxmax()

'Nintendo'

## 5 List out the sports by Konami Digital Entertainment

In [147]:
sp=games.groupby(['Genre'])

In [149]:
sp=sp.get_group('Sports')
sp

/var/folders/n8/js2blnd547x1vm4qzc05tq5c0000gn/T/ipykernel_2295/2840322743.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  sp=sp.get_group('Sports')


Rank                           Name Platform    Year   Genre  \
0          1                     Wii Sports      Wii  2006.0  Sports   
3          4              Wii Sports Resort      Wii  2009.0  Sports   
13        14                        Wii Fit      Wii  2007.0  Sports   
14        15                   Wii Fit Plus      Wii  2009.0  Sports   
77        78                        FIFA 16      PS4  2015.0  Sports   
...      ...                            ...      ...     ...     ...   
16576  16579              Rugby Challenge 3     XOne  2016.0  Sports   
16578  16581  Outdoors Unleashed: Africa 3D      3DS  2011.0  Sports   
16579  16582              PGA European Tour      N64  2000.0  Sports   
16581  16584                      Fit & Fun      Wii  2011.0  Sports   
16587  16590       Mezase!! Tsuri Master DS       DS  2009.0  Sports   

                  Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
0                  Nintendo     41.49     29.02      3.77         8.46   
3                  Nintendo     15.75     11.01      3.28         2.96   
13                 Nintendo      8.94      8.03      3.60         2.15   
14                 Nintendo      9.09      8.59      2.53         1.79   
77          Electronic Arts      1.11      6.06      0.06         1.26   
...                     ...       ...       ...       ...          ...   
16576  Alternative Software      0.00      0.01      0.00         0.00   
16578               Mastiff      0.01      0.00      0.00         0.00   
16579            Infogrames      0.01      0.00      0.00         0.00   
16581               Unknown      0.00      0.01      0.00         0.00   
16587           Hudson Soft      0.00      0.00      0.01         0.00   

       Global_Sales  
0             82.74  
3             33.00  
13            22.72  
14            22.00  
77             8.49  
...             ...  
16576          0.01  
16578          0.01  
16579          0.01  
16581          0.01  
16587          0.01  

[2346 rows x 11 columns]

In [153]:
sp[sp['Publisher']=='Konami Digital Entertainment']

Rank                                               Name Platform  \
249      250          Winning Eleven: Pro Evolution Soccer 2007      PS2   
292      293                      World Soccer Winning Eleven 9      PS2   
315      316        World Soccer Winning Eleven 8 International      PS2   
348      349                          Pro Evolution Soccer 2008      PS2   
474      475        World Soccer Winning Eleven 6 International      PS2   
...      ...                                                ...      ...   
16418  16421  Tennis no Oji-Sama: Doubles no Oji-Sama - Boys...       DS   
16420  16423          Winning Eleven: Pro Evolution Soccer 2007       PC   
16426  16429                               ESPN NBA 2Night 2002       XB   
16437  16440              The Cages: Pro Style Batting Practice      Wii   
16558  16561                          Pro Evolution Soccer 2008       PC   

         Year   Genre                     Publisher  NA_Sales  EU_Sales  \
249    2006.0  Sports  Konami Digital Entertainment      0.10      2.39   
292    2005.0  Sports  Konami Digital Entertainment      0.12      2.26   
315    2004.0  Sports  Konami Digital Entertainment      0.16      1.89   
348    2007.0  Sports  Konami Digital Entertainment      0.05      0.00   
474    2002.0  Sports  Konami Digital Entertainment      0.12      1.26   
...       ...     ...                           ...       ...       ...   
16418  2009.0  Sports  Konami Digital Entertainment      0.00      0.00   
16420  2006.0  Sports  Konami Digital Entertainment      0.00      0.01   
16426  2002.0  Sports  Konami Digital Entertainment      0.01      0.00   
16437  2010.0  Sports  Konami Digital Entertainment      0.01      0.00   
16558  2007.0  Sports  Konami Digital Entertainment      0.00      0.01   

       JP_Sales  Other_Sales  Global_Sales  
249        1.05         0.86          4.39  
292        0.90         0.77          4.06  
315        1.12         0.68          3.85  
348        0.64         2.93          3.63  
474        1.16         0.45          2.99  
...         ...          ...           ...  
16418      0.01         0.00          0.01  
16420      0.00         0.00          0.01  
16426      0.00         0.00          0.01  
16437      0.00         0.00          0.01  
16558      0.00         0.00          0.01  

[280 rows x 11 columns]

##  6 The game that has played on ps2 

In [241]:
ps2=games.groupby(['Platform'])

In [269]:
c=ps2.get_group('PS2')

/var/folders/n8/js2blnd547x1vm4qzc05tq5c0000gn/T/ipykernel_2295/2919429639.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  c=ps2.get_group('PS2')


In [278]:
c

Rank                                              Name Platform  \
17        18                     Grand Theft Auto: San Andreas      PS2   
24        25                       Grand Theft Auto: Vice City      PS2   
28        29                            Gran Turismo 3: A-Spec      PS2   
38        39                              Grand Theft Auto III      PS2   
47        48                                    Gran Turismo 4      PS2   
...      ...                                               ...      ...   
16559  16562      Sugar + Spice! Anoko no Suteki na Nanimokamo      PS2   
16561  16564                                    Kanokon: Esuii      PS2   
16572  16575                  Scarlett: Nichijou no Kyoukaisen      PS2   
16580  16583                                         Real Rode      PS2   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   

         Year      Genre                    Publisher  NA_Sales  EU_Sales  \
17     2004.0     Action         Take-Two Interactive      9.43      0.40   
24     2002.0     Action         Take-Two Interactive      8.41      5.49   
28     2001.0     Racing  Sony Computer Entertainment      6.85      5.09   
38     2001.0     Action         Take-Two Interactive      6.99      4.51   
47     2004.0     Racing  Sony Computer Entertainment      3.01      0.01   
...       ...        ...                          ...       ...       ...   
16559  2008.0  Adventure                    Alchemist      0.00      0.00   
16561  2008.0  Adventure                          5pb      0.00      0.00   
16572  2008.0  Adventure              Kadokawa Shoten      0.00      0.00   
16580  2008.0  Adventure              Kadokawa Shoten      0.00      0.00   
16595  2008.0     Racing                   Activision      0.00      0.00   

       JP_Sales  Other_Sales  Global_Sales  
17         0.41        10.57         20.81  
24         0.47         1.78         16.15  
28         1.87         1.16         14.98  
38         0.30         1.30         13.10  
47         1.10         7.53         11.66  
...         ...          ...           ...  
16559      0.01         0.00          0.01  
16561      0.01         0.00          0.01  
16572      0.01         0.00          0.01  
16580      0.01         0.00          0.01  
16595      0.00         0.00          0.01  

[2161 rows x 11 columns]

##  7 list out the adventure games

In [280]:
adv=games.groupby('Genre')

In [282]:
adv

In [311]:
c=adv.get_group('Adventure')

In [313]:
c

Rank                                Name Platform    Year      Genre  \
50        51  Super Mario Land 2: 6 Golden Coins       GB  1992.0  Adventure   
158      159                    Assassin's Creed     X360  2007.0  Adventure   
218      219                    Assassin's Creed      PS3  2007.0  Adventure   
251      252     Zelda II: The Adventure of Link      NES  1987.0  Adventure   
400      401          Rugrats: Search For Reptar       PS  1998.0  Adventure   
...      ...                                 ...      ...     ...        ...   
16572  16575    Scarlett: Nichijou no Kyoukaisen      PS2  2008.0  Adventure   
16575  16578               Neo Angelique Special      PSP  2008.0  Adventure   
16580  16583                           Real Rode      PS2  2008.0  Adventure   
16586  16589      Secret Files 2: Puritas Cordis       DS  2009.0  Adventure   
16591  16594                 Myst IV: Revelation       PC  2004.0  Adventure   

             Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
50            Nintendo      6.16      2.04      2.69         0.29   
158            Ubisoft      3.28      1.65      0.07         0.55   
218            Ubisoft      1.91      2.00      0.09         0.83   
251           Nintendo      2.19      0.50      1.61         0.08   
400                THQ      1.63      1.53      0.00         0.18   
...                ...       ...       ...       ...          ...   
16572  Kadokawa Shoten      0.00      0.00      0.01         0.00   
16575       Tecmo Koei      0.00      0.00      0.01         0.00   
16580  Kadokawa Shoten      0.00      0.00      0.01         0.00   
16586      Deep Silver      0.00      0.01      0.00         0.00   
16591          Ubisoft      0.01      0.00      0.00         0.00   

       Global_Sales  
50            11.18  
158            5.55  
218            4.83  
251            4.38  
400            3.34  
...             ...  
16572          0.01  
16575          0.01  
16580          0.01  
16586          0.01  
16591          0.01  

[1286 rows x 11 columns]

##  8 retrive the game with highest Sales in Europe

In [362]:
gd=games['EU_Sales'].max()

In [364]:
gd

29.02

##  9 List out the top 10 highest sales in japan on 2004

In [374]:
hi=games.groupby(['Year'])

In [382]:
hig=hi.get_group(2004)
hig

/var/folders/n8/js2blnd547x1vm4qzc05tq5c0000gn/T/ipykernel_2295/1404695354.py:1: FutureWarning: When grouping with a length-1 list-like, you will need to pass a length-1 tuple to get_group in a future version of pandas. Pass `(name,)` instead of `name` to silence this warning.
  hig=hi.get_group(2004)


Rank                               Name Platform    Year  \
17        18      Grand Theft Auto: San Andreas      PS2  2004.0   
47        48                     Gran Turismo 4      PS2  2004.0   
58        59  Pokemon FireRed/Pokemon LeafGreen      GBA  2004.0   
59        60                     Super Mario 64       DS  2004.0   
79        80                             Halo 2       XB  2004.0   
...      ...                                ...      ...     ...   
16330  16333      Ultraman Fighting Evolution 3      PS2  2004.0   
16362  16365                         Powerdrome       XB  2004.0   
16375  16378               Colin McRae Rally 04       PC  2004.0   
16445  16448        Wade Hixton's Counter Punch      GBA  2004.0   
16591  16594                Myst IV: Revelation       PC  2004.0   

              Genre                    Publisher  NA_Sales  EU_Sales  \
17           Action         Take-Two Interactive      9.43      0.40   
47           Racing  Sony Computer Entertainment      3.01      0.01   
58     Role-Playing                     Nintendo      4.34      2.65   
59         Platform                     Nintendo      5.08      3.11   
79          Shooter       Microsoft Game Studios      6.82      1.53   
...             ...                          ...       ...       ...   
16330      Fighting                    Banpresto      0.00      0.00   
16362        Racing                Evolved Games      0.01      0.00   
16375        Racing                  Codemasters      0.00      0.01   
16445        Sports    Destination Software, Inc      0.01      0.00   
16591     Adventure                      Ubisoft      0.01      0.00   

       JP_Sales  Other_Sales  Global_Sales  
17         0.41        10.57         20.81  
47         1.10         7.53         11.66  
58         3.15         0.35         10.49  
59         1.25         0.98         10.42  
79         0.05         0.08          8.49  
...         ...          ...           ...  
16330      0.01         0.00          0.01  
16362      0.00         0.00          0.01  
16375      0.00         0.00          0.01  
16445      0.00         0.00          0.01  
16591      0.00         0.00          0.01  

[763 rows x 11 columns]

In [406]:
hig.nlargest(10,"JP_Sales")

Rank                                           Name Platform    Year  \
180    181  Dragon Quest VIII: Journey of the Cursed King      PS2  2004.0   
58      59              Pokemon FireRed/Pokemon LeafGreen      GBA  2004.0   
130    131                        Pokémon Emerald Version      GBA  2004.0   
1110  1112             Dragon Quest V: Tenkuu no Hanayome      PS2  2004.0   
1351  1353          Classic NES Series: Super Mario Bros.      GBA  2004.0   
59      60                                 Super Mario 64       DS  2004.0   
686    688                             WarioWare Touched!       DS  2004.0   
315    316    World Soccer Winning Eleven 8 International      PS2  2004.0   
47      48                                 Gran Turismo 4      PS2  2004.0   
1445  1447                               Samurai Warriors      PS2  2004.0   

             Genre                     Publisher  NA_Sales  EU_Sales  \
180   Role-Playing                   Square Enix      0.65      0.75   
58    Role-Playing                      Nintendo      4.34      2.65   
130   Role-Playing                      Nintendo      2.57      1.58   
1110  Role-Playing              Enix Corporation      0.00      0.00   
1351      Platform                      Nintendo      0.00      0.00   
59        Platform                      Nintendo      5.08      3.11   
686         Puzzle                      Nintendo      0.52      0.45   
315         Sports  Konami Digital Entertainment      0.16      1.89   
47          Racing   Sony Computer Entertainment      3.01      0.01   
1445        Action               Electronic Arts      0.22      0.05   

      JP_Sales  Other_Sales  Global_Sales  
180       3.61         0.20          5.21  
58        3.15         0.35         10.49  
130       2.06         0.21          6.41  
1110      1.65         0.00          1.65  
1351      1.39         0.03          1.43  
59        1.25         0.98         10.42  
686       1.21         0.13          2.31  
315       1.12         0.68          3.85  
47        1.10         7.53         11.66  
1445      1.06         0.02          1.35

##  10  List out the lowest sales in all countries on 2004

In [419]:
hig.nsmallest(1,["JP_Sales",'NA_Sales','Other_Sales','EU_Sales','Global_Sales'])

Rank                   Name Platform    Year   Genre Publisher  \
16118  16121  Football Manager 2005       PC  2004.0  Sports      Sega   

       NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
16118       0.0      0.01       0.0          0.0          0.01

## 11 In which all platforms Grand Theft Auto: San Andreas	was played

In [463]:
act=games[games['Name']=='Grand Theft Auto: San Andreas'][['Name','Platform']]
act

Name Platform
17    Grand Theft Auto: San Andreas      PS2
873   Grand Theft Auto: San Andreas       XB
2120  Grand Theft Auto: San Andreas       PC
9827  Grand Theft Auto: San Andreas     X360